core/src/main/java/edu/wpi/grip/core/events/RenderEvent.java

```diff
@@ -0,0 +1,4 @@
+package edu.wpi.grip.core.events;
+
+public class RenderEvent {
```

Comment:
You know that we could avoid using `runAndWait` or anything similar by providing permits to read socket values from the pipeline. You could acquire a permit for the socket value, read the value, copy it into your own local variable and then let it go.
This would even work if we ever used the async event bus.

I don't know if this idea is completely convoluted.